extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java

```diff
@@ -19,13 +19,34 @@ public static OidcTenantConfig provider(OidcTenantConfig.Provider provider) {
             case LINKEDIN -> linkedIn();
             case MASTODON -> mastodon();
             case MICROSOFT -> microsoft();
+            case SLACK -> slack();
             case SPOTIFY -> spotify();
             case STRAVA -> strava();
             case TWITCH -> twitch();
             case TWITTER, X -> twitter();
         };
     }
 
+    private static OidcTenantConfig slack() {
+        OidcTenantConfig ret = new OidcTenantConfig();
+        ret.setAuthServerUrl("https://slack.com");
+        ret.setApplicationType(OidcTenantConfig.ApplicationType.WEB_APP);
+        ret.getCredentials().getClientSecret().setMethod(Method.POST);
+
+        var token = ret.getToken();
+        token.setPrincipalClaim("name");
+        token.setSubjectRequired(true);
+
+        var authentication = ret.getAuthentication();
+        authentication.setScopes(List.of("email", "profile"));
+        authentication.setForceRedirectHttpsScheme(true);
+        authentication.setAddOpenidScope(true);
```

Comment:
Quarkus adds it itself if it is not configured

Reply:
If I explicitly know that Slack requires it https://api.slack.com/authentication/sign-in-with-slack  then making it explicit here makes sense to me. I'll drop it, but I think it is much better to configure "this is what I need" because noone will remember if for example default changes "under some condition"